statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,721 @@
+from __future__ import division
+
+__all__ = ["ZeroInflatedPoisson", "ZeroInflatedGeneralizedPoisson",
+           "ZeroInflatedNegativeBinomialP"]
+
+
+import numpy as np
+import statsmodels.base.model as base
+import statsmodels.base.wrapper as wrap
+import statsmodels.regression.linear_model as lm
+from statsmodels.discrete.discrete_model import (DiscreteModel, CountModel,
+                                                 Poisson, Logit, CountResults,
+                                                 L1CountResults, Probit,
+                                                 _discrete_results_docs,
+                                                 GeneralizedPoisson,
+                                                 NegativeBinomialP)
+from statsmodels.distributions import zipoisson, zigenpoisson, zinegbin
+from statsmodels.tools.numdiff import (approx_fprime, approx_hess,
+                                       approx_hess_cs, approx_fprime_cs)
+from statsmodels.tools.decorators import (resettable_cache, cache_readonly)
+
+
+class GenericZeroInflated(CountModel):
+    __doc__ = """
+    Generiz Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, exog_infl=None, offset=None,
+                 inflation='logit', exposure=None, missing='none', **kwargs):
+        super(GenericZeroInflated, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+
+        if exog_infl is None:
+            self.k_inflate = 1
+            self.exog_infl = np.ones((endog.size, self.k_inflate))
+        else:
+            self.exog_infl = exog_infl
+            self.k_inflate = exog_infl.shape[1]
+
+        if len(exog.shape) == 1:
+            self.k_exog = 1
+        else:
+            self.k_exog = exog.shape[1]
+
+        self.infl = inflation
+        if inflation == 'logit':
+            self.model_infl = Logit(np.zeros(self.exog_infl.shape[0]),
```

Comment:
just a comment:
endog=zeros is not used because we only need likelihood related methods.
using binary `(endog == 0).astype(float)` might be useful for estimating an extreme corner case (if prob(y=0|model_main) is small or close to zero)